src/pip/_internal/network/auth.py

```diff
@@ -219,7 +219,7 @@ def _prompt_for_password(self, netloc):
         if not username:
             return None, None
         auth = get_keyring_auth(netloc, username)
-        if auth:
+        if auth and auth[0] and auth[1]:
```

Comment:
I’d explicitly say `auth[0] is not None and auth[1] is not None` here. I *think* technically the password can be an empty string?

Reply:
Also, is there any possibility that auth could be something other than a sequence with at least 2 elements? Do we need to protect against IndexError? (I;d imagine only getting one value might be more likely than not getting a sequence at all, in reality).

Reply:
There are many places in the code base that relies on the return value being a 2-tuple, and we’d be in deep trouble if it doesn’t. I’m inclined to add a type hint on the function and move on.

Reply:
@uranusjr, yes that condition can be made explicit. However, we'd still need to check the presence of `auth`. Only then `auth[0]` and `auth[1]` would be present. In case, the password comes out to be an empty string, then the explicit condition would catch that case.

@pfmoore I don't think there would be a possibility for that. As the auth object is retrieved from here https://github.com/pypa/pip/blob/daff81124ab76138da323a5173b120dd88190f56/src/pip/_internal/network/auth.py#L46

And this either returns `None` or  a pair of `username` and `password`

Reply:
Yeah, I meant `if auth and auth[0] is not None and auth[1] is not None:` here. Sorry for the confusion 🙂